src/main/kotlin/calculator/domain/Calculator.kt

```diff
@@ -0,0 +1,25 @@
+package calculator.domain
+
+class Calculator(private val expression: Expression) {
+
+    fun calculate(): Int {
+        if (expression.value.isEmpty()) {
+            return 0
+        }
+
+        val operands = expression.value.mapNotNull { it as? ExpressionElement.OperandElement }
+        val operators = expression.value.mapNotNull { it as? ExpressionElement.OperatorElement }
+
+        return operands.reduceIndexed { index, acc, operandElement ->
+            ExpressionElement.OperandElement(
+                operators[index - BUFFER_OPERATOR]
+                    .value
+                    .calculator(acc.value, operandElement.value)
+            )
```

Comment:
지금 같은 경우에도 operator 객체에게 메시지를 보내볼 수 있을 것 같아요. 😃

예를들자면 아래같은 방식이 될 것 같아요. 🤔
```suggestion
            operators[index - BUFFER_OPERATOR].calculate(acc, operandElement)
```